xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-default/src/main/java/org/xwiki/platform/security/requiredrights/internal/analyzer/XWikiDocumentRequiredRightAnalyzer.java

```diff
@@ -54,50 +50,31 @@ public class XWikiDocumentRequiredRightAnalyzer implements RequiredRightAnalyzer
     @Inject
     private DocumentContextExecutor documentContextExecutor;
 
-    @Inject
-    @Named("translation")
-    private BlockSupplierProvider<String> translationMessageSupplierProvider;
-
-    @Inject
-    private RequiredRightAnalyzer<XDOM> xdomRequiredRightAnalyzer;
-
     @Inject
     private RequiredRightAnalyzer<BaseObject> objectRequiredRightAnalyzer;
 
     @Inject
     private RequiredRightAnalyzer<BaseClass> classRequiredRightAnalyzer;
 
     @Inject
-    private VelocityDetector velocityDetector;
+    private Provider<XWikiContext> contextProvider;
 
     @Inject
-    private Provider<XWikiContext> contextProvider;
+    @Named("content")
+    private RequiredRightAnalyzer<XWikiDocument> contentRequiredRightAnalyzer;
 
     @Override
     public List<RequiredRightAnalysisResult> analyze(XWikiDocument document) throws RequiredRightsException
     {
         // Analyze the content
         try {
+            ArrayList<RequiredRightAnalysisResult> result =
```

Comment:
Doesn't SonarQube complain about the use of a concrete type on the left-hand side of the assignment?

Reply:
At least locally, I don't see any warning. But I'm changing it to `List`, I think this is `ArrayList` because that's the automatic type that you get when you use the `.var` refactoring in IntelliJ.